core/src/main/scala/org/scalarelational/extra/HikariSupport.scala

```diff
@@ -8,27 +8,26 @@ import org.scalarelational.model.SQLDatastore
  */
 trait HikariSupport extends SQLDatastore {
   // Automatically converts DataSources to be wrapped by HikariDataSource
-  dataSourceProperty.change.on {
-    case evt => evt.newValue match {
-      case null => // Ignore null DataSource
-      case ds: HikariDataSource => // Ignore HikariDataSource
-      case ds => {
-        val config = new HikariConfig()
-        config.setDataSource(ds)
-        val hikari = new HikariDataSource(config)
-        hikari.suspendPool()
-        dataSourceProperty := hikari
-      }
-    }
+  dataSourceProperty.filterCycles.attach {
```

Comment:
``filterCycles`` does exactly what its name says. When we do ``dataSourceProperty := hikari`` later, the handler will not be called (because it would lead to cyclic behaviour). In your version you dealt with this by providing an explicit ``case`` block checking for ``HikariDataSource`` values.